lib/translations_checker/checker.rb

```diff
@@ -0,0 +1,113 @@
+require "translations_checker/diff"
+require "translations_checker/issue"
+require "translations_checker/issues_presenter"
+require "translations_checker/terminal_colours"
+require "translations_checker/indentation"
+require "translations_checker/git_diff"
+
+module TranslationsChecker
+  class Checker
+    include TerminalColours
+
+    using Indentation
+
+    def file_issues(en_diff)
+      translated_locales.flat_map { |locale| check_translation(locale, en_diff) }
+    end
+
+    def issues
+      @issues ||= diffs_for_locale("en").flat_map(&method(:file_issues))
+    end
+
+    # :reek:TooManyStatements
+    # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+    def run
+      if issues.empty?
+        puts green("No translations issues detected.")
+        return
+      end
+
+      puts yellow("Please correct the following translations issues before pushing:")
+      puts
+      puts IssuesPresenter.new(diffs_for_locale("en").flat_map(&method(:file_issues)))
+      puts
+      puts example
+      puts
+      puts yellow("Don't forget to keep the .yml keys hierarchy\n")
+      exit 1
+    end
+    # rubocop:enable Metrics/AbcSize, Metrics/MethodLength
+
+    def self.run
+      new.run
+    end
+
+    private
+
+    def example
+      <<~EXAMPLE
+        ===========================
+        Example:
+
+          config/locales/en/en.yml:
+            en:
+              new: New
+
+          config/locales/ja/ja.yml:
+            ja:
+              new: [ja] New
+        ===========================
+      EXAMPLE
+    end
+
+    def diffs_for_locale(locale)
+      diffs.select { |diff| diff.locale == locale }
+    end
+
+    def diff_for_locale_file(locale_file)
+      diffs.detect { |diff| diff.locale_file == locale_file } || EmptyDiff.new(locale_file)
+    end
+
+    def check_deleted_change(en_change, locale_change)
+      NotDeleted.new(en_change, locale_change) unless locale_change.deleted?
+    end
+
+    # :reek:NilCheck, :reek:FeatureEnvy
+    # rubocop:disable Metrics/MethodLength
+    def check_change(en_change, locale_change)
```

Comment:
this seems a bit complicated but unsure how to refactor

Reply:
Yeah, hence the exclusions. I have at least one planned change that might help resolve this particular mess.

Reply:
Possibly move check_change to a change checker class?
```
deleted_change
missing_new_key
unchanged_key
incorrect_format
```

Reply:
Good idea